pkg/controller/che/che_controller.go

```diff
@@ -549,49 +560,52 @@ func (r *ReconcileChe) Reconcile(request reconcile.Request) (reconcile.Result, e
 		}
 	}
 
-	cheWorkspaceSA, err := deploy.SyncServiceAccountToCluster(deployContext, "che-workspace")
-	if cheWorkspaceSA == nil {
-		logrus.Info("Waiting on service account 'che-workspace' to be created")
-		if err != nil {
-			logrus.Error(err)
-		}
-		if !tests {
-			return reconcile.Result{RequeueAfter: time.Second}, err
-		}
-	}
+	// Openshift oAuth client handles permissions for workspaces itself.
+	if !util.IsOAuthEnabled(instance) && !util.IsWorkspaceInSameNamespaceWithChe(instance) {
+		policies := append(getCheCreateNamespacesPolicy(), getCheManageNamespacesPolicy()...)
 
-	// create exec role for CheCluster server and workspaces
-	execRole, err := deploy.SyncExecRoleToCluster(deployContext)
-	if execRole == nil {
-		logrus.Info("Waiting on role 'exec' to be created")
+		deniedRules, err := r.permissionChecker.GetNotPermittedPolicyRules(policies, "")
 		if err != nil {
-			logrus.Error(err)
-		}
-		if !tests {
 			return reconcile.Result{RequeueAfter: time.Second}, err
 		}
-	}
+		// fall back to the "narrower" workspace namespace strategy
+		if len(deniedRules) > 0 {
+			if _, ok := instance.Spec.Server.CustomCheProperties["CHE_INFRA_KUBERNETES_NAMESPACE_DEFAULT"]; ok {
+				delete(instance.Spec.Server.CustomCheProperties, "CHE_INFRA_KUBERNETES_NAMESPACE_DEFAULT")
```

Comment:
```suggestion
instance.Spec.Server.CustomCheProperties = delete(instance.Spec.Server.CustomCheProperties, "CHE_INFRA_KUBERNETES_NAMESPACE_DEFAULT")
```

Reply:
delete removes elements directly in the map and returns void. Thanks I will remove

```
if _, ok := instance.Spec.Server.CustomCheProperties["CHE_INFRA_KUBERNETES_NAMESPACE_DEFAULT"]; ok {
```